src/httpserver/http_request.hpp

```diff
@@ -240,27 +241,27 @@ class http_request {
      /**
       * Default constructor of the class. It is a specific responsibility of apis to initialize this type of objects.
      **/
-     http_request() = default;
+     http_request() : cache(std::make_unique<data_cache>()) {}
 
      http_request(MHD_Connection* underlying_connection, unescaper_ptr unescaper):
          underlying_connection(underlying_connection),
-         unescaper(unescaper) { }
+         unescaper(unescaper), cache(std::make_unique<data_cache>()) {}
 
+     http_request(const http_request& b) = delete;
      /**
-      * Copy constructor.
-      * @param b http_request b to copy attributes from.
+      * Move constructor.
+      * @param b http_request b to move attributes from.
      **/
-     http_request(const http_request& b) = default;
      http_request(http_request&& b) noexcept = default;
 
-     http_request& operator=(const http_request& b) = default;
+     http_request& operator=(const http_request& b) = delete;
      http_request& operator=(http_request&& b) = default;
 
      ~http_request();
 
      std::string path;
      std::string method;
-     std::map<std::string, std::string, http::arg_comparator> args;
+     http::arg_map args;
```

Comment:
We should just remove this, as it isn't used and just muddles the waters on where the cache really is.